script.js

```diff
@@ -0,0 +1,746 @@
+            // console.log("-- out --");
+            event.stopPropagation();
+            updateHoverText(e.currentTarget, "leave");
+        });
+
+        // ======= updateHoverText =======
+        function updateHoverText(hoverEl, enterLeave) {
+            // console.log("updateHoverText");
+            var hoverText;
+            var region = $(hoverEl).parents().eq(1).attr('id');
+            if (enterLeave == "enter") {
+                if ($(hoverEl).hasClass('buffer')) {
+                    if (app.state.selRegions[region]) {
+                        $(hoverEl).addClass("entered");
+                        if ($(hoverEl).hasClass('ft500')) {
+                            hoverText = "500 ft buffer";
+                        } else if ($(hoverEl).hasClass('ft1000')) {
+                            hoverText = "1000 ft buffer";
+                        } else if ($(hoverEl).hasClass('ft2500')) {
+                            hoverText = "2500 ft buffer";
+                        } else if ($(hoverEl).hasClass('ft5280')) {
+                            hoverText = "1 mile buffer";
+                        }
+                    }
+                } else {
+                    if ($(hoverEl).hasClass('region')) {
+                        hoverText = regions[$(hoverEl).closest('tr').attr('id')].name;
+                    } else if ($(hoverEl).hasClass('start')) {
+                        hoverText = "map click start loction";
+                    } else if ($(hoverEl).hasClass('end')) {
+                        hoverText = "map click end loction";
+                    } else if (($(hoverEl).attr('id') == 'clearAll-r') && ($(hoverEl).text() == 'all')) {
+                        hoverText = "select all regions";
+                    } else if (($(hoverEl).attr('id') == 'clearAll-r') && ($(hoverEl).text() == 'clear')) {
+                        hoverText = "clear all regions/buffers";
+                    }
+                }
+            } else {
+                if ($(hoverEl).hasClass('buffer')) {
+                    if (app.state.selRegions[region]) {
+                        $(hoverEl).removeClass("entered");
+                    }
+                }
+                hoverText = ".";
+            }
+            $("#hoverText").text(hoverText);
+        }
+    },
+
+    // ======= toggleFilterState =======
+    toggleFilterState: function(toggleEl) {
+        console.log("toggleFilterState");
+        var region = $(toggleEl).parents().eq(1).attr('id');
+        var buffer = null;
+        console.log("  app.state.selRegions[region]1: ", app.state.selRegions[region]);
+
+        // == update region filter
+        if ($(toggleEl).hasClass('region')) {
+
+            // == toggle region state
+            app.state.selRegions[region] ?
+                app.state.selRegions[region] = false :
+                app.state.selRegions[region] = true;
+
+            // == update clear/all button text
```

Comment:
Remove comments from lines 269-278 from this PR. This looks like experimental code you're working on. This can be in a separate [work in progress](http://ben.balter.com/2015/12/08/types-of-pull-requests/#work-in-progress-wip) PR 😄  that's what branches are for!